tests/mmm/test_delayed_saturated_mmm.py

```diff
@@ -506,3 +513,115 @@ def mock_property(self):
+                "beta_channel": {
+                    "dist": "HalfNormal",
+                    "kwargs": {"sigma": np.array([0.4533017, 0.25488063])},
+                },
+                "alpha": {
+                    "dist": "Beta",
+                    "kwargs": {
+                        "alpha": np.array([3, 3]),
+                        "beta": np.array([3.55001301, 2.87092431]),
+                    },
+                },
+                "lam": {
+                    "dist": "Gamma",
+                    "kwargs": {
+                        "alpha": np.array([3, 3]),
+                        "beta": np.array([4.12231653, 5.02896872]),
+                    },
+                },
+                "likelihood": {
+                    "dist": "StudentT",
+                    "kwargs": {"nu": 3, "sigma": 2},
+                },
+                "gamma_control": {"dist": "Normal", "kwargs": {"mu": 0, "sigma": 2}},
+                "gamma_fourier": {"dist": "Laplace", "kwargs": {"mu": 0, "b": 1}},
+            },
+        ],
+        ids=["default_config", "custom_config"],
+    )
+    def test_model_config(
+        self, model_config: Dict, toy_X: pd.DataFrame, toy_y: pd.Series
+    ):
+        # Create model instance with specified config
+        model = DelayedSaturatedMMM(
+            date_column="date",
+            channel_columns=["channel_1", "channel_2"],
+            adstock_max_lag=2,
+            yearly_seasonality=2,
+            model_config=model_config,
+        )
+
+        model.build_model(X=toy_X, y=toy_y.to_numpy())
+        # Check for default configuration
+        if model_config is None:
+            # assert observed RV type, and priors of some/all free_RVs.
+            assert isinstance(
+                model.model.observed_RVs[0].owner.op, pm.Normal
+            )  # likelihood
+            # Add more asserts as needed for default configuration
+
+        # Check for custom configuration
+        else:
+            # assert custom configuration is applied correctly
+            assert isinstance(
+                model.model.observed_RVs[0].owner.op, pm.StudentT
+            )  # likelihood
+            assert isinstance(
+                model.model["beta_channel"].owner.op, pm.HalfNormal
+            )  # beta_channel
+
+
+def test_get_valid_distribution(mmm):
+    normal_dist = mmm._get_distribution({"dist": "Normal"})
+    assert normal_dist is pm.Normal
+
+
+def test_get_invalid_distribution(mmm):
+    with pytest.raises(ValueError, match="does not exist in PyMC"):
+        mmm._get_distribution({"dist": "NonExistentDist"})
+
+
+def test_invalid_likelihood_type(mmm):
+    with pytest.raises(
+        ValueError,
+        match=r"The distribution used for the likelihood is not allowed\. Please, use one of the following distributions: \['Normal', 'StudentT', 'Laplace', 'Logistic', 'LogNormal', 'Wald', 'TruncatedNormal', 'Gamma', 'AsymmetricLaplace', 'VonMises'\]\.",
```

Comment:
You don't need to match the whole error message. This way the test will still work if you simply support more dists in the future

```suggestion
        match="The distribution used for the likelihood is not allowed",
```